llpc/builder/llpcBuilderImplImage.cpp

```diff
@@ -984,6 +999,49 @@ Value* BuilderImplImage::CreateImageLoad(
                                   instName);
     }
 
+    // For 64-bit texel, only the first component is loaded, other components are filled in with (0, 0, 1). This
+    // operation could be viewed as supplement of the intrinsic call.
+    if (is64Bit && pOrigTexelTy->isVectorTy())
+    {
+        Value* pTexel = pResult;
+        if (isa<StructType>(pResultTy))
+        {
+            pTexel = CreateExtractValue(pResult, uint64_t(0));
+        }
+        pTexel = CreateBitCast(pTexel, getDoubleTy()); // Casted to double
+
+        Value* pZero = ConstantFP::get(getDoubleTy(), 0.0);
+        Value* pOne  = ConstantFP::get(getDoubleTy(), 1.0);
+        if (pOrigTexelTy->isIntOrIntVectorTy())
+        {
+            pZero = CreateBitCast(getInt64(0), getDoubleTy());
```

Comment:
Can you add a comment explaining why this is needed? It could seems that you end up with the same constants either way.

Reply:
I add this because the OpenGL spec assumes the default values are (0,0,0,1) while the values are either FP or integer typed. Thus, you see a bitcast from i64 to double for the following handling. The default values are encoded as i64 while LLVM requires us to represent it as double type to call image intrinsic. This is somewhat confusing and I will simplify the logic.